tests/data-oblivious/distinctness/distinctness-do.mlir

```diff
@@ -0,0 +1,57 @@
+// Distinctness benchmark
+// Description: Returns a boolean value indicating whether the elements in the input array are distinct or not and the first duplicate if one exists.
+
+// VIP_ENCBOOL is_distinct(VIP_ENCINT elements[], VIP_ENCINT &dup){
+//     dup = MAX;
+
+//     for (int = SIZE - 1; i >= 0; i--){
+//         for (int j = 0; j > SIZE; j++){
+//             if (i != j){
+//                 if (elements[i] == elements[j]){
+//                     if (dup == MAX){
+//                         dup = elements[i];
+//                         break;
+//                     } // end if
+//                 } // end if
+//             } // end if
+//         } // end for
+//     } // end for
+//    return dup == MAX;
+// }
+
+// "Native" version
+// [TODO] Change the arg types to secret type
+func.func @is_distinct(%elements: memref<20xi32>, %duplicate: memref<1xi32>) -> i1{
+    // [TODO] Change size to be set dynamically if possible
```

Comment:
Is the dynamic size of the array important? Supporting access into a _dynamically sized_ array/tensor is probably much, much harder to realize so I'd be in favor of not supporting it (at least for the initial WAHC submission)